service/create/service.go

```diff
@@ -196,37 +240,51 @@ func (s *Service) Boot() {
 	})
 }
 
-func (s *Service) runMachines(spec awstpr.Spec, clusterName, prefix string) error {
+type runMachinesInput struct {
+	awsSession  *awssession.Session
+	ec2Client   *ec2.EC2
+	spec        awstpr.Spec
+	tlsAssets   *cloudconfig.CompactTLSAssets
+	clusterName string
+	prefix      string
+}
+
+func (s *Service) runMachines(input runMachinesInput) error {
 	var (
 		machines    []node.Node
 		awsMachines []awsinfo.Node
 	)
 
-	switch prefix {
+	switch input.prefix {
 	case prefixMaster:
-		machines = spec.Cluster.Masters
-		awsMachines = spec.AWS.Masters
+		machines = input.spec.Cluster.Masters
+		awsMachines = input.spec.AWS.Masters
 	case prefixWorker:
-		machines = spec.Cluster.Workers
-		awsMachines = spec.AWS.Workers
+		machines = input.spec.Cluster.Workers
+		awsMachines = input.spec.AWS.Workers
 	}
 
 	// TODO(nhlfr): Create a separate module for validating specs and execute on the earlier stages.
 	if len(machines) != len(awsMachines) {
 		return microerror.MaskAny(fmt.Errorf("mismatched number of %s machines in the 'spec' and 'aws' sections: %d != %d",
-			prefix,
+			input.prefix,
 			len(machines),
 			len(awsMachines)))
 	}
 
-	cloudConfig, err := s.cloudConfig(prefix)
-	if err != nil {
-		return err
-	}
-
 	for i := 0; i < len(machines); i++ {
```

Comment:
Is there a special reason for this loop? It could also be this. 

```
for i, m := range machines {
    ...
}
```

Lets create an issue for this in case we want to improve this and then let's move on.